src/components/MaPagination/MaPagination.vue

```diff
@@ -1,47 +1,164 @@
 <template>
     <a-pagination
+        class="ma-pagination"
         v-bind="$attrs"
-        :page-size-options="[20, 30, 50, 100, 200]"
-        :current-page="pagination.currentPage"
-        :page-size="pagination.pageSize"
-        :total="pagination.total"
+        :page-size-options="[10, 20, 30, 50, 100, 200]"
+        :page-size="pageSize"
+        :current="current"
+        :total="total"
+        :show-quick-jumper="showQuickJumper"
+        :show-size-changer="showSizeChanger"
+        :disabled="disabled"
+        :size="size"
+        :show-total="this.showTotalFunction"
         v-on="$listeners"
         @change="onPageChange"
-        @show-size-change="onPageSizeChange"
+        @showSizeChange="onPageSizeChange"
     />
 </template>
 
 <script>
     import Pagination from 'ant-design-vue/lib/pagination/index.js';
-
     export default {
         name: 'ma-pagination',
         components: {
             APagination: Pagination,
         },
         props: {
-            pagination: { type: Object, default: () => ({}) },
+            pageSize: {
+                type: Number,
+                default: 10,
+            },
+            pageSizeOptions: {
+                type: Array,
+                default: () => [10, 20, 30, 50, 100, 200],
+            },
+            total: {
+                type: Number,
+                default: 0,
+            },
+            current: {
+                type: Number,
+                default: 1,
+            },
+            showQuickJumper: {
+                type: Boolean,
+                default: false,
+            },
+            showSizeChanger: {
+                type: Boolean,
+                default: false,
+            },
+            showTotal: {
+                type: Boolean,
+                default: false,
+            },
+            customShowTotal: {
+                type: Function,
+                default: null,
+            },
+            disabled: {
+                type: Boolean,
+                default: false,
+            },
+            size: {
+                type: String,
+                default: '',
+            },
         },
-        data() {
-            return {
-                current: 2,
-            };
+        computed: {
+            showTotalFunction() {
```

Comment:
showTotal is enough definetion
```suggestion
            showTotal() {
```